### DISCUSSION TITLE
Questions about first setup

### DISCUSSION THREAD

**Discussion by Pustiu** [open-meteo/open-meteo#601] [General]
Hi,

I came across this system a few weeks ago and would like to use it in a system I have been working on for some time. It is an internal application, the application takes weather data from different sources and calculates the electricity production for photovoltaic and wind farms. 
I start by giving you some technical data and then maybe you can help me with some answers to some questions.

I only need hourly and 15 minutes forecast.
For the hour I need: [https://api.open-meteo.com/v1/forecast?latitude=44.428&longitude=26.0967&hourly=temperature_2m,relative_humidity_2m,dew_point_2m,apparent_temperature,precipitation_probability,weather_code,surface_pressure,cloud_cover,cloud_cover_low,cloud_cover_mid,cloud_cover_high,wind_speed_10m,wind_direction_10m,wind_gusts_10m,sunshine_duration,shortwave_radiation,direct_radiation,diffuse_radiation,shortwave_radiation_instant,direct_radiation_instant,diffuse_radiation_instant](https://api.open-meteo.com/v1/forecast?latitude=44.428&longitude=26.0967&hourly=temperature_2m,relative_humidity_2m,dew_point_2m,apparent_temperature,precipitation_probability,weather_code,surface_pressure,cloud_cover,cloud_cover_low,cloud_cover_mid,cloud_cover_high,wind_speed_10m,wind_direction_10m,wind_gusts_10m,sunshine_duration,shortwave_radiation,direct_radiation,diffuse_radiation,shortwave_radiation_instant,direct_radiation_instant,diffuse_radiation_instant)

For 15 minutes: [https://api.open-meteo.com/v1/forecast?latitude=44.428&longitude=26.0967&minutely_15=temperature_2m,relative_humidity_2m,dew_point_2m,apparent_temperature,precipitation,sunshine_duration,weather_code,wind_speed_10m,wind_direction_10m,wind_gusts_10m,shortwave_radiation,direct_radiation,diffuse_radiation,shortwave_radiation_instant,direct_radiation_instant,diffuse_radiation_instant](https://api.open-meteo.com/v1/forecast?latitude=44.428&longitude=26.0967&minutely_15=temperature_2m,relative_humidity_2m,dew_point_2m,apparent_temperature,precipitation,sunshine_duration,weather_code,wind_speed_10m,wind_direction_10m,wind_gusts_10m,shortwave_radiation,direct_radiation,diffuse_radiation,shortwave_radiation_instant,direct_radiation_instant,diffuse_radiation_instant)

Weather data I want to get for 250 locations in Romania and a few from other countries in Europe (less than 20 locations). I will make requests for these locations hour by hour.

I set up OpenMeteo on a VPS:
CPU: 12 vCPU Cores
RAM: 48 GB RAM
STORAGE: 400 GB NVMe

I tried to set up a few crons to update the weather data periodically but I didn't succeed, some of the mentioned fields appear with null value.

If you can help me with some crons to pull data for the mentioned fields, in what order they should be executed and at what time interval. Possibly just the sources needed for Europe. Data older than the current day I don't need and I want to delete them.

I saw that there are two ways to preload weather data: "openmeteo-api sync" and "openmeteo-api download".  Which is the best configuration solution.

Regardless of the answers be sure that I will purchase at least one license in the next period because I want to support this project. I want the VPS to be just a fallback in case open-meteo.com is unavailable.

Thank you very much!

_Sorry for my English_

**Comment by guidocioni** [comment]
Without going into the details of your specific issue please do note that 15 mins data are coming from  the ICON-D2 model, which does **not** cover Romania. 

![icon_d2_modellgebiet](https://github.com/open-meteo/open-meteo/assets/12760310/bd46f0f5-eb41-40f2-be6c-0996cae8fccd)

Maybe this is part of the problem? 
If you're requesting 15 minutely data from locations outside of this domain hourly data will be used and interpolated to 15 minutes.

**Reply by Pustiu** [reply]
I managed to sync all the data except: surface_pressure
I run first: openmeteo-api sync copernicus_dem90 static
And after:
`openmeteo-api sync dwd_icon,dwd_icon_eu,dwd_icon_d2,dwd_icon_d2_15min,ncep_gfs013,ncep_gfs025,ncep_hrrr_conus,ncep_hrrr_conus_15min,meteofrance_arpege_world025,meteofrance_arpege_europe,meteofrance_arome_france0025,meteofrance_arome_france_hd,ecmwf_ifs04,cmc_gem_gdps,metno_nordic_pp,cma_grapes_global,bom_access_global temperature_2m,dew_point_2m,apparent_temperature,precipitation,weather_code,surface_pressure,cloud_cover,cloud_cover_low,cloud_cover_mid,cloud_cover_high,wind_speed_10m,wind_direction_10m,wind_gusts_10m,sunshine_duration,shortwave_radiation,direct_radiation,diffuse_radiation,shortwave_radiation_instant,direct_radiation_instant,diffuse_radiation_instant,pressure_msl`

But I think there's too much, too many domains that aren't necessary.
I'm trying to get the same data on my server as on open-meteo. Yes, I know that ICON-D2 does not provide data for Romania

**Reply by patrick-zippenfenig** [reply]
Domains `dwd_icon,dwd_icon_eu,dwd_icon_d2,dwd_icon_d2_15min` and variables `temperature_2m,dew_point_2m,precipitation,weather_code,surface_pressure,cloud_cover,cloud_cover_low,cloud_cover_mid,cloud_cover_high,wind_u_component_10m,wind_v_component_10m,wind_gusts_10m,shortwave_radiation,direct_radiation,diffuse_radiation,pressure_msl` should be enough.

As mentioned in the docs, many variables are derived from other variables. E.g. wind speed from U/V components

**Reply by Pustiu** [reply]
Hi Patrick,
I'm syncing by the domains and fields you indicated but on: relative_humidity_2m,dew_point_2m, apparent_temperature and precipitation_probability I get null

**Reply by patrick-zippenfenig** [reply]
Ah, sorry. ICON already provides `relative_humidity_2m` directly. https://openmeteo.s3.amazonaws.com/index.html#data/dwd_icon_d2/

For precipitation probability you need the variable `precipitation_probability` from the model `ncep_gefs025_probability` https://openmeteo.s3.amazonaws.com/index.html#data/ncep_gefs025_probability/
